rpc/src/endpoint/tx_search.rs

```diff
@@ -36,21 +36,23 @@ impl Request {
     }
 }
 
-impl crate::Request for Request {
-    type Response = Response;
-
+impl RequestMessage for Request {
     fn method(&self) -> Method {
         Method::TxSearch
     }
 }
 
-impl crate::SimpleRequest for Request {}
+impl<S: Dialect> crate::Request<S> for Request {
+    type Response = Response<S::Event>;
+}
+
+impl<S: Dialect> crate::SimpleRequest<S> for Request {}
 
 #[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct Response {
-    pub txs: Vec<tx::Response>,
+pub struct Response<Ev> {
```

Comment:
This is annoying to adapt to in client code, while the data type the client has to deal with does not really have to be different: the only problem we're solving with the generics is differences in serialization. I'm inclined to get this back to `Response` using domain types, with `DialectResponse<Ev>` alongside for JSON deserialization. This, however, will require changes to the `Request` trait.